tests/enhancer/dsp/test_filter.py

```diff
@@ -0,0 +1,47 @@
+"""Tests for enhancer.dsp.filter module"""
+
+import pytest
+import torch
+
+from clarity.enhancer.dsp.filter import AudiometricFIR
+
+SAMPLE_RATE = 44100
+
+
+@pytest.fixture
+def use_torch():
```

Comment:
I think this is me not being familiar enough with pytest 😄  We need to execute these lines for every test that uses torch, or calls torch code -- but I didn't want the code floating in global space as that seemed inelegant - i.e. also run if the file was imported.  I've now set the 'autouse=True' so that it doesn't need to appear as an unused parameter.  Perhaps it belongs in conftest ? Except most of our modules don't use torch

Reply:
I'm by no means well versed in `pytest` either!

I thought to use a fixture it had to be provided as an argument to a subsequent `def test_something(my_fixture):` The `autouse=True` option seems sensible here though it does what is says on the tin and I'd agree with _not_ moving that to `tests/conftest.py` as only a handful of tests require torch configuration.

I can see why there is the need to setup a consistent set of parameters for Torch so the results are reproducible, but I'm even less familiar with Torch than I am with others aspects of Python and am unsure if these settings are set once globally and then used each and every time a new object of Torch is instantiated (kind of like how logging works in Python with a global logger which took me a while to get my head round) or if they need explicitly stating each time, but then I don't see how the settings get passed into each of the tests here.

But if it works and gives consistent results then that is all good. :+1: My understanding can catch up in due course :laughing: !